fuel/converters/ilsvrc2010.py

```diff
@@ -0,0 +1,826 @@
+
+    Returns
+    -------
+    synsets : ndarray, 1-dimensional, compound dtype
+        A table containing ILSVRC2010 metadata for the "synonym sets"
+        or "synsets" that comprise the classes and superclasses,
+        including the following fields:
+         * `ILSVRC2010_ID`: the integer ID used in the original
+           competition data.
+         * `WNID`: A string identifier that uniquely identifies
+           a synset in ImageNet and WordNet.
+         * `wordnet_height`: The length of the longest path to
+           a leaf nodein the FULL ImageNet/WordNet hierarchy
+           (leaf nodes in the FULL ImageNet/WordNet hierarchy
+           have `wordnet_height` 0).
+         * `gloss`: A string representation of an English
+           textual description of the concept represented by
+           this synset.
+         * `num_children`: The number of children in the hierarchy
+           for this synset.
+         * `words`: A string representation, comma separated,
+           of different synoym words or phrases for the concept
+           represented by this synset.
+         * `children`: A vector of `ILSVRC2010_ID`s of children
+           of this synset, padded with -1. Note that these refer
+           to `ILSVRC2010_ID`s from the original data and *not*
+           the zero-based index in the table.
+         * `num_train_images`: The number of training images for
+           this synset.
+    cost_matrix : ndarray, 2-dimensional, uint8
+        A 1000x1000 matrix containing the precomputed pairwise
+        cost (based on distance in the hierarchy) for all
+        low-level synsets (i.e. the thousand possible output
+        classes with training data associated).
+
+    """
+    mat = loadmat(meta_mat, squeeze_me=True)
+    synsets = mat['synsets']
+    cost_matrix = mat['cost_matrix']
+    new_dtype = numpy.dtype([
+        ('ILSVRC2010_ID', numpy.int16),
+        ('WNID', ('S', max(map(len, synsets['WNID'])))),
+        ('wordnet_height', numpy.int8),
+        ('gloss', ('S', max(map(len, synsets['gloss'])))),
+        ('num_children', numpy.int8),
+        ('words', ('S', max(map(len, synsets['words'])))),
+        ('children', (numpy.int8, max(synsets['num_children']))),
+        ('num_train_images', numpy.uint16)
+    ])
+    new_synsets = numpy.empty(synsets.shape, dtype=new_dtype)
+    for attr in ['ILSVRC2010_ID', 'WNID', 'wordnet_height', 'gloss',
+                 'num_children', 'words', 'num_train_images']:
+        new_synsets[attr] = synsets[attr]
+    children = [numpy.atleast_1d(ch) for ch in synsets['children']]
+    padded_children = [
+        numpy.concatenate((c,
+                           -numpy.ones(new_dtype['children'].shape[0] - len(c),
+                                       dtype=numpy.int16)))
+        for c in children
+    ]
+    new_synsets['children'] = padded_children
+    return new_synsets, cost_matrix
+
+
+def extract_patch_images(f, which_set=None):
```

Comment:
Why not make `which_set` an `arg`, since you explicitly require that it is either `'train'`, `'valid'` or `'test'`?